src/sinks/util/sink.rs

```diff
@@ -493,12 +504,16 @@ where
         // or the batch got submitted by polling_complete above.
         let mut batch = self.batch.fresh();
 
-        batch.push(item);
-        self.set_linger(partition.clone());
+        match batch.push(item) {
+            PushResult::Full(item) => Ok(AsyncSink::NotReady(item)),
```

Comment:
Same here I think.

Reply:
This should actually be unreachable unless it's an unusable configuration. The batch was just created fresh the line before, so the push shouldn't report full.

Reply:
If that's the case I'd prefer we either panic or reorganize to remove this branch.

Reply:
I set it as `unreachable!`